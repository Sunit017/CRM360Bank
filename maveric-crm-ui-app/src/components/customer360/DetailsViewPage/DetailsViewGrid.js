import { Grid, Box } from "@mui/material";
import styled from "styled-components";

const Label = styled.div`
  marginright: 5px;
  display: flex;
  font-weight: 700;
  color: #68077b;
  // border: 1px solid red;
`;

const DetailsViewGrid = ({ viewDetails, resData }) => {
  const listColumn = viewDetails?.children;

  return (
    <>
      <Box
        data-testid="DetailsViewGrid"
        sx={{
          width: "100%",
          // maxWidth: 300,
          position: "relative",
          overflow: "auto",
          maxHeight: "calc(100vh - 200px)",
          borderRadius: "none",
          "&::-webkit-scrollbar": {
            width: 4,
            height: 4,
          },
          "&::-webkit-scrollbar-track": {
            backgroundColor: "#ced4da",
          },
          "&::-webkit-scrollbar-thumb": {
            backgroundColor: "#68077b",
            borderRadius: 100,
          },
        }}
      >
        <Grid
          container
          spacing={{ xs: 2, md: 3, lg: 1 }}
          columns={{ xs: 4, sm: 8, md: 12 }}
          data-testid="test-2"
        >
          {resData && Object.keys(resData).length>0
            ? Object.keys(resData).map((itmKey, index) => {
                const labelItem = listColumn.filter(
                  (item) => item.id === itmKey
                )[0];
                // const label = labelItem?.label || camelCaseToTitleCase(itmKey);
                const label = labelItem?.label
                return itmKey !== "addressId" && itmKey !== "relationshipId" && label ? (
                  <Grid item xs={2} sm={4} md={4} key={itmKey}>
                    <Box>
                      <Label>{label} :</Label>
                      {resData && resData[itmKey].toString()}
                    </Box>
                  </Grid>
                ) : null;
              })
            : (listColumn || []).map((item, index) => {
                const { id, label } = item;
                return (
                  <Grid item xs={2} sm={4} md={4} key={id}>
                    <Box>
                      <Label>{label} :</Label>
                      {resData && resData[id]}
                    </Box>
                  </Grid>
                );
              })}
        </Grid>
      </Box>
    </>
  );
};
export default DetailsViewGrid;
