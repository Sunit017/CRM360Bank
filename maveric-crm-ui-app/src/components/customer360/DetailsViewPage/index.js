import React, { useEffect, useState } from "react";
import HelmetPage from "../../common/HelmetPage";
import { useSelector } from "react-redux";
import DetailsViewGrid from "./DetailsViewGrid";
import styled from "styled-components";
import ViewPageList from "./ViewPageList";
import {
  getCustomerRelationshipApi,
  getCustomerNomineeApi,
  getCustomersById,
  getCustomerKycDetailsById,
  getCustomerPersonalDetailsById,
  getCustomerBusinessDetailsById,
  getCustomerFinancialDetailsById,
  getCustomerAMLDetailsById,
  getCustomerAddressDetailsById,
  getCustomerIdentificationDetailsById,
  getCustomerSpouseDetailsById,
  getCustomerNomineeDetailsById,
  getCustomerEmploymentDetailsById,
  getCustomerClientDetailsById,
  getCustomerIdMissionDetailsById,
  getCustomerLoanDetailsById,
  getCustomerIdmissiondedupeDetailsById,
} from "../../../services/Customer360Service";

const ViewDetailsTitle = styled.div`
  text-align: center;
  font-weight: 700;
  height: 25px;
`;
const DetailsViewPage = ({ currentMenu }) => {
  const [pageData, setPageData] = useState(null);
  // const [activeMenu, setActiveMenu] = useState("");
  const [pageDetails, setPageDetails] = useState([]);
  const customerSelection = useSelector((state) => state.customerSelection);

  const isDataValid=(data,key)=>data && data[key]
  useEffect(() => {
    const { customerId, IDM_ARN_NO, data } = customerSelection || {};
    let menuItem = "";
    if (currentMenu) {
      menuItem = currentMenu;
    } else {
      if (customerId) {
        menuItem = "customerDetails";
      } else if (IDM_ARN_NO) {
        menuItem = "personalDetails";
      }
    }
    const custId = data?.partyId;
    // setActiveMenu(menuItem);
    setPageData(null);
    if (menuItem === "customerDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      // setPageData(customerSelection?.data[0])
      isDataValid(data,"customerId")&& getCustomersById(data.customerId,setPageData)
    }

    if (menuItem === "personalDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerPersonalDetailsById(data.customerId,setPageData)
    }

    if (
      menuItem === "permanentAddress" ||
      menuItem === "employmentAddress" ||
      menuItem === "businessAddress"
    ) {
      const type = ViewPageList.filter((item) => item.id === menuItem)[0].addressType;
      // customerGetAddressApi(setPageData, type, custId);
      isDataValid(data,"customerId") && getCustomerAddressDetailsById(data.customerId,type.toString().toUpperCase(),setPageData)
    }

    if (menuItem === "businessDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerBusinessDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "employmentDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerEmploymentDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "spouseDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      // const custType = ViewPageList.filter(
      //   (item) => item.id === menuItem
      // )[0].label.split(" ");
      // getCustomerRelationshipApi(setPageData, "Spouse", custId); //1000000008
      isDataValid(data,"customerId")&& getCustomerSpouseDetailsById(data.customerId,setPageData)
    }
    if (menuItem === "nomineeDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      // getCustomerNomineeApi(setPageData,custId)//1000000008
      isDataValid(data,"customerId")&& getCustomerNomineeDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "identityDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId") && getCustomerIdentificationDetailsById(data.customerId,setPageData)
      // isDataValid(data,"customerId")&& getCustomerPersonalDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "financialDetails") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      
      isDataValid(data,"customerId")&& getCustomerFinancialDetailsById(data.customerId,setPageData)
      
    }

    if (menuItem === "clientStatus") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerClientDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "idMission") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerIdMissionDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "idMissionDedupe") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerIdmissiondedupeDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "AMLStatus") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerAMLDetailsById(data.customerId,setPageData) 
    }

    if (menuItem === "KYCHighRiskfields") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerKycDetailsById(data.customerId,setPageData)
    }

    if (menuItem === "loanFields") {
      // getCustomerPageApi(setPageData, menuItem, custId);
      isDataValid(data,"customerId")&& getCustomerLoanDetailsById(data.customerId,setPageData)
    }

    const listView = ViewPageList.filter((item) => item.id === menuItem);
    
    setPageDetails(listView[0]);
  }, [customerSelection, currentMenu]);
  console.info("pageData ##", pageData);
  return (
    <>
      <HelmetPage title="Customer 360 page" />
      <ViewDetailsTitle>{pageDetails?.label} </ViewDetailsTitle>
      {pageDetails?.children && (
        <DetailsViewGrid viewDetails={pageDetails} resData={pageData} />
      )}
    </>
  );
};

export default DetailsViewPage;
