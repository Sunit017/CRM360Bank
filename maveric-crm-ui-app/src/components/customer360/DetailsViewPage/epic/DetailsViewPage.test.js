import DetailsViewPage from "..";
import renderWithRedux from "../../../../testUtil";
import { cleanup, fireEvent, screen } from "@testing-library/react";
afterEach(cleanup);

test("Render DetailsViewPage ", () => {
  const container = renderWithRedux(
    <DetailsViewPage currentMenu={"personalDetails"} />,
    {}
  );
  expect(container).toMatchSnapshot();
});
test("Render DetailsViewPage with store ", () => {
  const container = renderWithRedux(
    <DetailsViewPage currentMenu={"customerDetails"} />,
    {
      customerSelection: { customerId: 1234, IDM_ARN_NO: 342455 },
    }
  );
});
