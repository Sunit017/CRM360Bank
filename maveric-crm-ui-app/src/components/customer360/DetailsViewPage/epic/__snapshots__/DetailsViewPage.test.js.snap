// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DetailsViewPage  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsnTZi eXXpkq"
      >
        Personal Details
         
      </div>
      <div
        class="MuiBox-root css-1c17ife"
        data-testid="DetailsViewGrid"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
          data-testid="test-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Salutation
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Suffix
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Civil Status
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Place of Birth
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Nationality
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Years in Residence
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Preferred Mailing Address
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Number of Dependents
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                My Job (employed or self-employed)
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Marketing Consent
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Date time field
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Nominee Counter
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Customer ID Type
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Card Issuance Status
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                My Job Code(employed or self-employed)
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Data Privacy Agreement
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Record Edit by Customer
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                FATCA - W9 Form Details - ID Type
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Term Conditions
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Client Type
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                FATCA - Certification of US Non-US Status
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                Designation
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                IDM ARN No
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                ARN No
                 :
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-bczRLJ fHPAUt"
              >
                FATCA - W9 Form Details - ID Number
                 :
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsnTZi eXXpkq"
    >
      Personal Details
       
    </div>
    <div
      class="MuiBox-root css-1c17ife"
      data-testid="DetailsViewGrid"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
        data-testid="test-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Salutation
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Suffix
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Civil Status
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Place of Birth
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Nationality
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Years in Residence
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Preferred Mailing Address
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Number of Dependents
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              My Job (employed or self-employed)
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Marketing Consent
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Date time field
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Nominee Counter
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Customer ID Type
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Card Issuance Status
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              My Job Code(employed or self-employed)
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Data Privacy Agreement
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Record Edit by Customer
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              FATCA - W9 Form Details - ID Type
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Term Conditions
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Client Type
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              FATCA - Certification of US Non-US Status
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              Designation
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              IDM ARN No
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              ARN No
               :
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="sc-bczRLJ fHPAUt"
            >
              FATCA - W9 Form Details - ID Number
               :
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
